### PR TITLE
CLOUDP-47524: Create a canonical build script

### PR DESCRIPTION
**Pull request by BenElgar**
Also improves docker caching and pins alpine version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 FROM golang:1.12 AS builder
 WORKDIR /usr/src
+
+# Download and cache dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
 COPY . .
 
-RUN CGO_ENABLED=0 go build -a -installsuffix cgo -o bin/atlas-service-broker
```
Comment (BenElgar) [inline Dockerfile]: The `-a -installsuffix cgo` is no longer necessary as of Go 1.10: https://github.com/golang/go/issues/9344#issuecomment-69944514

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+RUN ./scripts/build-production-binary.sh bin/atlas-service-broker
 
 # Run stage uses a much smaller base image to run the prebuilt binary
-FROM alpine:latest
```
Comment (BenElgar) [inline Dockerfile]: Pinning the version helps us to have more reproducible builds.

**Comment by BenElgar** [comment]
Oops—gave this the wrong branch name. 🤦‍♂

File: Dockerfile (hunk 2 above)
Comment (mrgunior) [reply Dockerfile]: This is good actually. Using latest might not always be the best idea, instead a specific version that we know is suitable.

**Comment by BenElgar** [comment]
> I assume this is what you meant today to use the binary instead of rebuilding?

No, this is only a partial measure. I want to stop building the binary twice at all.

**Pull request merged**